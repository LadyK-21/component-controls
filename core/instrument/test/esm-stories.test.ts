import { loadStoriesTests } from './loadTestFiles';

describe('esm-stories', () => {
  loadStoriesTests(
    undefined,
    ['esm', 'stories'],
    //['template-bind.tsx']
  );
});
