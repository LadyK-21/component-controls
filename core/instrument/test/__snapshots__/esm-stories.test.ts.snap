// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`esm-stories named-object-export.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": "2020-07-06T20:55:12.000Z",
    "dateModified": "2020-07-06T20:55:12.000Z",
    "package": "a764e2f1f1f8cc23edccb9cfb301ec8e",
    "title": "Story",
  },
  "packages": Object {
    "a764e2f1f1f8cc23edccb9cfb301ec8e": Object {
      "dependencies": Object {
        "@babel/generator": "^7.12.5",
        "@babel/parser": "^7.12.5",
        "@babel/traverse": "^7.12.5",
        "@component-controls/core": "^2.10.4",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gitlog": "^4.0.4",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "isomorphic-git": "^1.8.0",
        "js-string-escape": "^1.0.1",
        "path": "^0.12.7",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "sloc": "^0.2.1",
        "typescript": "^4.0.5",
      },
      "devDependencies": Object {
        "@babel/types": "^7.12.6",
        "@component-controls/ts-markdown-docs": "^2.6.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "@types/sloc": "^0.2.0",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "a764e2f1f1f8cc23edccb9cfb301ec8e",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/named-object-export.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "2.10.4",
    },
  },
  "stories": Object {},
  "transformed": "export default { title: 'Story' };
export const myStory = {
  i: 1,
};
",
}
`;

exports[`esm-stories no-arguments.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": "2020-07-06T20:55:12.000Z",
    "dateModified": "2020-07-06T20:55:12.000Z",
    "package": "5f97f1689681bb75b22fb6973c4c8397",
    "title": "Story",
  },
  "packages": Object {
    "5f97f1689681bb75b22fb6973c4c8397": Object {
      "dependencies": Object {
        "@babel/generator": "^7.12.5",
        "@babel/parser": "^7.12.5",
        "@babel/traverse": "^7.12.5",
        "@component-controls/core": "^2.10.4",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gitlog": "^4.0.4",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "isomorphic-git": "^1.8.0",
        "js-string-escape": "^1.0.1",
        "path": "^0.12.7",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "sloc": "^0.2.1",
        "typescript": "^4.0.5",
      },
      "devDependencies": Object {
        "@babel/types": "^7.12.6",
        "@component-controls/ts-markdown-docs": "^2.6.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "@types/sloc": "^0.2.0",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "5f97f1689681bb75b22fb6973c4c8397",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/no-arguments.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "2.10.4",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 31,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "() => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = () => {};
",
}
`;

exports[`esm-stories no-story.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": "2020-07-06T20:55:12.000Z",
    "dateModified": "2020-07-06T20:55:12.000Z",
    "package": "947c20fa355bb677d067a09d46558900",
    "title": "Story",
  },
  "packages": Object {
    "947c20fa355bb677d067a09d46558900": Object {
      "dependencies": Object {
        "@babel/generator": "^7.12.5",
        "@babel/parser": "^7.12.5",
        "@babel/traverse": "^7.12.5",
        "@component-controls/core": "^2.10.4",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gitlog": "^4.0.4",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "isomorphic-git": "^1.8.0",
        "js-string-escape": "^1.0.1",
        "path": "^0.12.7",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "sloc": "^0.2.1",
        "typescript": "^4.0.5",
      },
      "devDependencies": Object {
        "@babel/types": "^7.12.6",
        "@component-controls/ts-markdown-docs": "^2.6.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "@types/sloc": "^0.2.0",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "947c20fa355bb677d067a09d46558900",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/no-story.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "2.10.4",
    },
  },
  "stories": Object {},
  "transformed": "export default { title: 'Story' };
const i = 1;
",
}
`;

exports[`esm-stories props-argument.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": "2020-07-06T20:55:12.000Z",
    "dateModified": "2020-07-06T20:55:12.000Z",
    "package": "c525e04e9beb03220dd4fa6d09af4958",
    "title": "Story",
  },
  "packages": Object {
    "c525e04e9beb03220dd4fa6d09af4958": Object {
      "dependencies": Object {
        "@babel/generator": "^7.12.5",
        "@babel/parser": "^7.12.5",
        "@babel/traverse": "^7.12.5",
        "@component-controls/core": "^2.10.4",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gitlog": "^4.0.4",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "isomorphic-git": "^1.8.0",
        "js-string-escape": "^1.0.1",
        "path": "^0.12.7",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "sloc": "^0.2.1",
        "typescript": "^4.0.5",
      },
      "devDependencies": Object {
        "@babel/types": "^7.12.6",
        "@component-controls/ts-markdown-docs": "^2.6.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "@types/sloc": "^0.2.0",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "c525e04e9beb03220dd4fa6d09af4958",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/props-argument.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "2.10.4",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 5,
              "line": 0,
            },
            "start": Object {
              "column": 0,
              "line": 0,
            },
          },
          "name": "props",
          "value": "props",
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 34,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "props => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = props => {};
",
}
`;

exports[`esm-stories three-levels-alias.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": "2020-07-06T20:55:12.000Z",
    "dateModified": "2020-07-06T20:55:12.000Z",
    "package": "3135bea8e9a3103e26e81e541800e043",
    "title": "Story",
  },
  "packages": Object {
    "3135bea8e9a3103e26e81e541800e043": Object {
      "dependencies": Object {
        "@babel/generator": "^7.12.5",
        "@babel/parser": "^7.12.5",
        "@babel/traverse": "^7.12.5",
        "@component-controls/core": "^2.10.4",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gitlog": "^4.0.4",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "isomorphic-git": "^1.8.0",
        "js-string-escape": "^1.0.1",
        "path": "^0.12.7",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "sloc": "^0.2.1",
        "typescript": "^4.0.5",
      },
      "devDependencies": Object {
        "@babel/types": "^7.12.6",
        "@component-controls/ts-markdown-docs": "^2.6.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "@types/sloc": "^0.2.0",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "3135bea8e9a3103e26e81e541800e043",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/three-levels-alias.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "2.10.4",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 31,
              "line": 0,
            },
            "start": Object {
              "column": 1,
              "line": 0,
            },
          },
          "name": undefined,
          "value": Array [
            Object {
              "loc": Object {
                "end": Object {
                  "column": 24,
                  "line": 0,
                },
                "start": Object {
                  "column": 9,
                  "line": 0,
                },
              },
              "name": "name",
              "value": Array [
                Object {
                  "loc": Object {
                    "end": Object {
                      "column": 16,
                      "line": 0,
                    },
                    "start": Object {
                      "column": 11,
                      "line": 0,
                    },
                  },
                  "name": "first",
                  "value": "first",
                },
                Object {
                  "loc": Object {
                    "end": Object {
                      "column": 22,
                      "line": 0,
                    },
                    "start": Object {
                      "column": 18,
                      "line": 0,
                    },
                  },
                  "name": "last",
                  "value": "last",
                },
              ],
            },
            Object {
              "loc": Object {
                "end": Object {
                  "column": 29,
                  "line": 0,
                },
                "start": Object {
                  "column": 26,
                  "line": 0,
                },
              },
              "name": "age",
              "value": "age",
            },
          ],
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 61,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "({ name: { first, last }, age }) => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = ({ name: { first, last }, age }) => {};
",
}
`;

exports[`esm-stories two-arguments.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": "2020-07-06T20:55:12.000Z",
    "dateModified": "2020-07-06T20:55:12.000Z",
    "package": "33a786db69e3366a4131eef973b8ad77",
    "title": "Story",
  },
  "packages": Object {
    "33a786db69e3366a4131eef973b8ad77": Object {
      "dependencies": Object {
        "@babel/generator": "^7.12.5",
        "@babel/parser": "^7.12.5",
        "@babel/traverse": "^7.12.5",
        "@component-controls/core": "^2.10.4",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gitlog": "^4.0.4",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "isomorphic-git": "^1.8.0",
        "js-string-escape": "^1.0.1",
        "path": "^0.12.7",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "sloc": "^0.2.1",
        "typescript": "^4.0.5",
      },
      "devDependencies": Object {
        "@babel/types": "^7.12.6",
        "@component-controls/ts-markdown-docs": "^2.6.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "@types/sloc": "^0.2.0",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "33a786db69e3366a4131eef973b8ad77",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/two-arguments.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "2.10.4",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 6,
              "line": 0,
            },
            "start": Object {
              "column": 1,
              "line": 0,
            },
          },
          "name": "props",
          "value": "props",
        },
        Object {
          "loc": Object {
            "end": Object {
              "column": 15,
              "line": 0,
            },
            "start": Object {
              "column": 8,
              "line": 0,
            },
          },
          "name": "context",
          "value": "context",
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 45,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "(props, context) => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = (props, context) => {};
",
}
`;

exports[`esm-stories two-levels-alias.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": "2020-07-06T20:55:12.000Z",
    "dateModified": "2020-07-06T20:55:12.000Z",
    "package": "244bd9e12bf9fbfc25a493133455fa30",
    "title": "Story",
  },
  "packages": Object {
    "244bd9e12bf9fbfc25a493133455fa30": Object {
      "dependencies": Object {
        "@babel/generator": "^7.12.5",
        "@babel/parser": "^7.12.5",
        "@babel/traverse": "^7.12.5",
        "@component-controls/core": "^2.10.4",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gitlog": "^4.0.4",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "isomorphic-git": "^1.8.0",
        "js-string-escape": "^1.0.1",
        "path": "^0.12.7",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "sloc": "^0.2.1",
        "typescript": "^4.0.5",
      },
      "devDependencies": Object {
        "@babel/types": "^7.12.6",
        "@component-controls/ts-markdown-docs": "^2.6.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "@types/sloc": "^0.2.0",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "244bd9e12bf9fbfc25a493133455fa30",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/two-levels-alias.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "2.10.4",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 21,
              "line": 0,
            },
            "start": Object {
              "column": 1,
              "line": 0,
            },
          },
          "name": undefined,
          "value": Array [
            Object {
              "loc": Object {
                "end": Object {
                  "column": 14,
                  "line": 0,
                },
                "start": Object {
                  "column": 9,
                  "line": 0,
                },
              },
              "name": "name",
              "value": "MyNam",
            },
            Object {
              "loc": Object {
                "end": Object {
                  "column": 19,
                  "line": 0,
                },
                "start": Object {
                  "column": 16,
                  "line": 0,
                },
              },
              "name": "age",
              "value": "age",
            },
          ],
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 51,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "({ name: MyNam, age }) => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = ({ name: MyNam, age }) => {};
",
}
`;

exports[`esm-stories two-levels-sub-arguments.js 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": "2020-07-06T20:55:12.000Z",
    "dateModified": "2020-07-06T20:55:12.000Z",
    "package": "dda8a6c0184dad7581d495667190312f",
    "title": "Story",
  },
  "packages": Object {
    "dda8a6c0184dad7581d495667190312f": Object {
      "dependencies": Object {
        "@babel/generator": "^7.12.5",
        "@babel/parser": "^7.12.5",
        "@babel/traverse": "^7.12.5",
        "@component-controls/core": "^2.10.4",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gitlog": "^4.0.4",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "isomorphic-git": "^1.8.0",
        "js-string-escape": "^1.0.1",
        "path": "^0.12.7",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "sloc": "^0.2.1",
        "typescript": "^4.0.5",
      },
      "devDependencies": Object {
        "@babel/types": "^7.12.6",
        "@component-controls/ts-markdown-docs": "^2.6.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "@types/sloc": "^0.2.0",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "dda8a6c0184dad7581d495667190312f",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/two-levels-sub-arguments.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "2.10.4",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 14,
              "line": 0,
            },
            "start": Object {
              "column": 1,
              "line": 0,
            },
          },
          "name": undefined,
          "value": Array [
            Object {
              "loc": Object {
                "end": Object {
                  "column": 7,
                  "line": 0,
                },
                "start": Object {
                  "column": 3,
                  "line": 0,
                },
              },
              "name": "name",
              "value": "name",
            },
            Object {
              "loc": Object {
                "end": Object {
                  "column": 12,
                  "line": 0,
                },
                "start": Object {
                  "column": 9,
                  "line": 0,
                },
              },
              "name": "age",
              "value": "age",
            },
          ],
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 44,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "({ name, age }) => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = ({ name, age }) => {};
",
}
`;

exports[`esm-stories typescript.ts 1`] = `
Object {
  "components": Object {},
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {},
    "date": "2020-07-06T20:55:12.000Z",
    "dateModified": "2020-07-06T20:55:12.000Z",
    "package": "b92edb32865b71e9d6623213bf5d565d",
    "title": "Story",
  },
  "packages": Object {
    "b92edb32865b71e9d6623213bf5d565d": Object {
      "dependencies": Object {
        "@babel/generator": "^7.12.5",
        "@babel/parser": "^7.12.5",
        "@babel/traverse": "^7.12.5",
        "@component-controls/core": "^2.10.4",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gitlog": "^4.0.4",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "isomorphic-git": "^1.8.0",
        "js-string-escape": "^1.0.1",
        "path": "^0.12.7",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "sloc": "^0.2.1",
        "typescript": "^4.0.5",
      },
      "devDependencies": Object {
        "@babel/types": "^7.12.6",
        "@component-controls/ts-markdown-docs": "^2.6.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "@types/sloc": "^0.2.0",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "b92edb32865b71e9d6623213bf5d565d",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/stories/typescript.ts",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "2.10.4",
    },
  },
  "stories": Object {
    "myStory": Object {
      "arguments": Array [
        Object {
          "loc": Object {
            "end": Object {
              "column": 18,
              "line": 0,
            },
            "start": Object {
              "column": 1,
              "line": 0,
            },
          },
          "name": "props",
          "value": "props",
        },
      ],
      "id": "myStory",
      "loc": Object {
        "end": Object {
          "column": 48,
          "line": 2,
        },
        "start": Object {
          "column": 23,
          "line": 2,
        },
      },
      "name": "myStory",
      "source": "(props: Properties) => {}",
    },
  },
  "transformed": "export default { title: 'Story' };
export const myStory = (props: Properties) => {};
",
}
`;
